.github/workflows/validate.yaml

```diff
@@ -35,9 +35,25 @@ jobs:
           sudo mv /usr/bin/yq /usr/local/bin/yq
 
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           path: ./src/github.com/${{ github.repository }}
+          fetch-depth: 0
+
+      - name: Merge upstream
+        if: github.event_name == 'pull_request'
+        working-directory: ./src/github.com/${{ github.repository }}
+        run: |
+          if ! git config user.name > /dev/null; then
+            git config user.name "John Doe"
+          fi
+          if ! git config user.email > /dev/null; then
+            git config user.email "johndoe@localhost"
+          fi
+          git remote add upstream https://github.com/${{ github.repository }}.git
+          git fetch upstream ${{ github.base_ref }}
```

Comment:
I'm curious if "git fetch" is required here? From `git help pull` command:
```
More precisely, git pull runs git fetch with the given parameters and then depending on configuration options
       or command line flags, will call either git rebase or git merge to reconcile diverging branches.
```

Reply:
The original code from which you copied had `--no-rebase --no-commit`. I suppose especially the first option should be specified (or `--rebase`) when the branches diverged. Again, from git help:
```
If the current branch and
       the remote have diverged, the user needs to specify how to reconcile the divergent branches with --rebase or
       --no-rebase
```

Reply:
No, it shouldn't be needed

Reply:
Removed https://github.com/openshift-knative/serverless-operator/pull/1789/commits/66654b8a1afff56405f7b2d70f2af28546e9742e

Reply:
Thanks! Added --no-rebase https://github.com/openshift-knative/serverless-operator/pull/1789/commits/83966da1c26466c499ebf427e43597c7ed3023a2

Reply:
The --no-commit is not possible because we use git to detect validate codegen